src/System.Private.CoreLib/src/Internal/Runtime/InteropServices/ComponentActivator.cs

```diff
@@ -0,0 +1,197 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Runtime.Loader;
+
+namespace Internal.Runtime.InteropServices
+{
+    public static class ComponentActivator
+    {
+        public enum StringFormat
+        {
+            UTF8 = 0,
+            UTF16 = 1,
+            Ansi = 2
+        };
+
+        [StructLayout(LayoutKind.Sequential)]
+        [CLSCompliant(false)]
+        public unsafe struct CreateNativeDelegateArgs
+        {
+            public IntPtr   assemblyPath;      // Fully qualified path to assembly
+            public IntPtr   delegateType;      // Assembly qualified delegate type name
+            public IntPtr   typeName;          // Assembly qualified type name
+            public IntPtr   methodName;        // public static method assignable to delegateType
+            public IntPtr   functionHandle;    // pointer where to store the function pointer result
+            public byte     stringFormat;      // String format passed as byte
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        public struct CreateDelegateArgs
+        {
+            public string    assemblyPath;      // Fully qualified path to assembly
+            public string    delegateType;      // Assembly qualified delegate type name
+            public string    typeName;          // Assembly qualified type name
+            public string    methodName;        // public static method assignable to delegateType
+        }
+
+        private static Dictionary<string, Assembly> s_Assemblies = new Dictionary<string, Assembly>();
```

Comment:
If we keep this, then please specify the string comparer to use for the keys.

Reply:
I wanted exact match which I assumed was the default.

Reply:
It probably is, but if I remember correctly there were security rules about specifying comparers explicitly always - to make it explicit and not an accident.

Reply:
> security rules about specifying comparers explicitly always

These rules are about case-insensitive comparisons (to avoid picking up current culture by accident). I do not think that there rules like this for exact match.

Reply:
I chose to use case insensitive compare on Windows.